receiver/http.go

```diff
@@ -80,6 +102,18 @@ func (rcvr receiver) answer(w http.ResponseWriter, r *http.Request, code int, in
 }
 
 func (rcvr receiver) handle(w http.ResponseWriter, r *http.Request) (int, error) {
+	if rcvr.auth != nil {
+		ok, err := rcvr.auth.auth(r)
+		if !ok && err == nil {
+			return 401, skogul.Error{Source: "http receiver", Reason: "Auth error"}
+		}
+
+		if !ok {
+			// !ok could be either "yeah no this didn't authenticate" or "yeah no this doesn't have access"...
+			return 401, err
+		}
+	}
```

Comment:
Suggestion: Make auth() accept the nil-value. E.g., remove alle the above lines so it only reads: if err := rcvr.auth.auth(r); err != nil { return 401, ....} .... Then have auth() do: if auth == nil { return nil }? It underlines the fact that there is no scenario where you get an error but that is ok, and makes handle() trivial.